src/towncrier/_project.py

```diff
@@ -5,17 +5,24 @@
 Responsible for getting the version and name from a project.
 """
 
-
 from __future__ import annotations
 
 import sys
 
 from importlib import import_module
+from importlib.metadata import version as metadata_version
 from types import ModuleType
+from typing import Any
 
 from incremental import Version as IncrementalVersion
 
 
+try:
+    from importlib.metadata import packages_distributions
+except ImportError:
+    from importlib_metadata import packages_distributions  # type: ignore
```

Comment:
`packages_distributions` is 3.10+, and since towncrier uses pyupgrade it seems a good idea to use a formation that pyupgrade can automatically remove in the future:

```suggestion
if sys.version_info >= (3, 10):
    from importlib.metadata import packages_distributions
else:
    from importlib_metadata import packages_distributions  # type: ignore
```